client/my-sites/google-my-business/stats/chart.js

```diff
@@ -0,0 +1,122 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { isEqual, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import CardHeading from 'components/card-heading';
+import FakeData from './fake-data';
+import PieChart from 'components/pie-chart';
+import PieChartLegend from 'components/pie-chart/legend';
+import SectionHeader from 'components/section-header';
+import { changeGoogleMyBusinessStatsInterval } from 'state/google-my-business/actions';
+import { getInterval } from 'state/google-my-business/selector';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+class GoogleMyBusinessStatsChart extends Component {
+	static propTypes = {
+		changeGoogleMyBusinessStatsInterval: PropTypes.func.isRequired,
+		chartTitle: PropTypes.oneOfType( [ PropTypes.func, PropTypes.string ] ),
+		data: PropTypes.array.isRequired,
+		dataSeriesInfo: PropTypes.object,
+		description: PropTypes.string,
+		interval: PropTypes.oneOf( [ 'week', 'month', 'quarter' ] ),
+		siteId: PropTypes.number.isRequired,
+		statType: PropTypes.string.isRequired,
+		title: PropTypes.string.isRequired,
+	};
+
+	static defaultProps = {
+		dataSeriesInfo: {},
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			transformedData: this.transformData( props.data ),
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.data !== nextProps.data ) {
+			this.setState( {
+				transformedData: this.transformData( nextProps.data ),
+			} );
+		}
+	}
+
+	shouldComponentUpdate( nextProps ) {
+		//@TODO: Once the data comes from redux, re-evaluate the need for deep equal
+		return (
+			this.props.interval !== nextProps.interval || ! isEqual( this.props.data, nextProps.data )
+		);
```

Comment:
Please correct me if I'm wrong, but my understanding is that Redux's [`connect()` function](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) will perform shallow comparisons.

Reply:
I will investigate that and get back to you.

Reply:
when you 
`connect( ( state ) => ( { bla: state.bla } )( Component )`
if `this.props.bla !== nextProps.bla` then it’s a different bla, something changed - reducer returned a new instance of the object.

That’s how redux works, the reducer must return a new instance
if they are equal, it’s the same object, the reducer returned the same instance.

If `this.props.data` comes from redux, it's enough for us to compare references.